sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.controller.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  ForbiddenException,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from "@nestjs/common";
+import { IInstitutionUserToken } from "../../auth/userToken.interface";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  IsInstitutionAdmin,
+  UserToken,
+} from "../../auth/decorators";
+import { DesignationAgreementService, FormService } from "../../services";
+import { getUTCNow } from "../../utilities";
+import {
+  GetDesignationAgreementDto,
+  SubmitDesignationAgreementDto,
+} from "./models/designation-agreement.model";
+import { InstitutionUserRoles } from "../../auth/user-types.enum";
+import { FormNames } from "../../services/form/constants";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("institution/designation-agreement")
+export class DesignationAgreementController {
+  constructor(
+    private readonly designationAgreementService: DesignationAgreementService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Initiates a new designation agreement request. This action
+   * is meant to me initiated by the institution signing officer
+   * for further assessment of the Ministry.
+   * @param institutionId institution id requesting the designation.
+   * @returns the new designation agreement id created.
+   */
+  @IsInstitutionAdmin()
+  @Post()
+  async submitDesignationAgreement(
+    @UserToken() userToken: IInstitutionUserToken,
+    @Body() payload: SubmitDesignationAgreementDto,
+  ) {
+    // Validates if the user has the right role.
+    const isLegalSigningAuthority = userToken.authorizations.hasAdminRole(
```

Comment:
isLegalSigningAuthority is check only on this API alone. Otherwise a decorator would be nice here IMO.

Reply:
I thought about the decorator and decided to use the "inline" check because apparently there will be no other endpoint that requires the same check. The only purpose of the LegalSigningAuthority is to execute this submission, all other access for visualization, for instance, are for any Admin. If there is any scenario in the future that would required the same check maybe we can create one.
Every time that we introduce a new decorator globally it will be tested for every single request for every single endpoint.
So, I can still create one if that is a Team decision 😉

Reply:
My point is we need decorator only if LegalSigningAuthority is needed for any other API in future which can be done if the requirement comes. At present what you have done is good.